subsys/bluetooth/host/hci_core.c

```diff
@@ -3387,6 +3398,25 @@ static int le_init(void)
 		net_buf_unref(rsp);
 	}
 
+#if defined(CONFIG_BT_BROADCASTER)
+	if (IS_ENABLED(CONFIG_BT_EXT_ADV) && BT_DEV_FEAT_LE_EXT_ADV(bt_dev.le.features)) {
+		/* Read LE Max Adv Data Len */
+		err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_READ_MAX_ADV_DATA_LEN, NULL, &rsp);
+		if (err == 0) {
+			le_read_maximum_adv_data_len_complete(rsp);
+			net_buf_unref(rsp);
+		} else if (err == -EIO) {
+			LOG_WRN("Controller does not support 'LE_READ_MAX_ADV_DATA_LEN'. "
+				"Assuming maximum length is 31 bytes.");
+			bt_dev.le.max_adv_data_len = 31;
```

Comment:
Minor: Since we are now checking the EXT_ADV bits before issuing the command, perhaps it should be a LOG_ERR as that is clearly a non-conformant controller
```suggestion
	if (IS_ENABLED(CONFIG_BT_EXT_ADV) && BT_DEV_FEAT_LE_EXT_ADV(bt_dev.le.features)) {
		/* Read LE Max Adv Data Len */
		err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_READ_MAX_ADV_DATA_LEN, NULL, &rsp);
		if (err == 0) {
			le_read_maximum_adv_data_len_complete(rsp);
			net_buf_unref(rsp);
		} else if (err == -EIO) {
			LOG_ERR("Controller does not support 'LE_READ_MAX_ADV_DATA_LEN'. "
				"Assuming maximum length is 31 bytes.");
			bt_dev.le.max_adv_data_len = 31;
```

The alternative would to stop the initialization here, as we can't be sure that the controller will act correctly with future commands (i.e. `return err` even if `err == -EIO`).

Reply:
I kinda agree with you but at the same time this isn't a "unrecoverable error" right? We should define how we want to use the different log level in the Bluetooth code guidelines